src/cli.rs

```diff
@@ -129,9 +129,37 @@ pub fn parse_args_from(mut args: Vec<OsString>) -> crate::Result<ParsedArgs> {
 
             let command = Command::Decompress { files, output_folder };
             ParsedArgs { command, flags }
-        },
+        }
         _ => unreachable!("You should match each subcommand passed."),
     };
 
     Ok(parsed_args)
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::process::Command;
+
+    fn gen_args(text: &str) -> Vec<OsString> {
+        let args = text.split_whitespace();
+        args.map(OsString::from).collect()
+    }
+
+    #[test]
+    fn test_something() {
+        let args = gen_args("ouch foo.zip bar.zip");
+        let result = parse_args_from(args).unwrap(); // ERROR: file `ouch` not found
```

Comment:
Well spotted issue!

We are making checks inside of the `parse_args_from` which make testing much harder, I will have to refactor this code and separate functionality in a way that let's us test it even with files that are not present (error occurs when trying to `.canonicalize()` on paths).